custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js

```diff
@@ -76,38 +76,172 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
+                    'stylingClass' : 'medium-col'
+                },
+                {
+                    'mData' : 'thr',
+                    'heading' : 'Take Home Ration (21+ days)',
+                    'tooltipValue' : 'Of the total number of pregnant women, lactating women (0-6 months children) and 6-36 months children enrolled for Anganwadi services, the percentage of pregnant women, lactating women (0-6 months children) and 6-36 months children who were provided THR for at least 21 days in the current month.',
+                    'columnValueType' : 'percentage',
+                    'columnValueIndicator' : 'thr',
+                    'stylingClass' : 'medium-col'
+                },
+            ],
+        },
+        'children' : [
+            {
+                'mData' : 'sn',
+                'heading' : 'Supplementary Nutrition (21+ days)',
+                'tooltipValue' : 'Of the total children between 3-6 years of age and enrolled for Anganwadi services, the percentage of children who were provided Hot Cooked Meal i.e. supplementary nutrition for at least 21 days in the current month.',
+                'columnValueType' : 'percentage',
+                'columnValueIndicator' : 'supNutrition',
+                'stylingClass' : 'medium-col'
+            },
+            {
+                'mData' : 'pse',
+                'heading' : 'Pre-school Education (21+ days)',
+                'tooltipValue' : 'Of the total children between 3-6 years of age and enrolled for Anganwadi services, the percentage of children who attended Pre-school education for at least 21 days in the current month.',
+                'columnValueType' : 'percentage',
+                'columnValueIndicator' : 'pse',
+                'stylingClass' : 'medium-col'
+            },
+            {
+                'mData' : 'gm',
+                'heading' : 'Growth Monitoring',
+                'tooltipValue' : 'Of the total children between <b>3-5 years</b> of age and enrolled for Anganwadi services, the percentage of children who were weighed in the current month.<br><br><b>Growth Monitoring is done only for children till 5 years of age.</b>',
+                'columnValueType' : 'percentage',
+                'columnValueIndicator' : 'gm36',
+                'stylingClass' : 'medium-col'
+            },
+        ]
+    };
+
+    vm.getSddTableData = function (isPwLwChildren, isAwc) {
+        return isPwLwChildren ?
+            (isAwc ? vm.sddTableData['pw_lw_children']['awc'] : vm.sddTableData['pw_lw_children']['non-awc']) :
+            vm.sddTableData['children'];
+    };
+
+    vm.isPwLwChildrenTab = function () {
+        return ($location.path().indexOf('/pw_lw_children') !== -1);
+    };
+
+    vm.isAwcDataShown = function () {
+        // if sector level is selected --> selectedLocationLevel is 3
+        return (parseInt($location.search()['selectedLocationLevel'], 10) === 3);
```

Comment:
what does the `10` do here?

Reply:
its a radix/base for parsing string to integer. (eg: to convert it to binary it will be 2, for decimal it will be 10)